yearn/contracts/SaddleStrategy.sol

```diff
@@ -0,0 +1,396 @@
+}
+
+/// @notice Interface for the LPRewards contracts.
+/// @dev This is an interface with just a few functions. For more info and
+///      function description please see:
+///      https://github.com/keep-network/keep-ecdsa/blob/main/solidity/contracts/LPRewards.sol
+interface ILPRewards {
+    function stake(uint256 amount) external;
+
+    function withdraw(uint256 amount) external;
+
+    function getReward() external;
+
+    function setGated(bool _gated) external;
+
+    function balanceOf(address account) external view returns (uint256);
+
+    function wrappedToken() external view returns (address);
+}
+
+/// @notice Interface for the Uniswap v2 router.
+/// @dev This is an interface with just a few function signatures of the
+///      router contract. For more info and function description please see:
+///      https://uniswap.org/docs/v2/smart-contracts/router02.
+interface IUniswapV2Router {
+    function swapExactTokensForTokens(
+        uint256,
+        uint256,
+        address[] calldata,
+        address,
+        uint256
+    ) external;
+}
+
+/// @title SaddleStrategy
+/// @notice This strategy is meant to be used with the Saddle tBTC v2 pool vault.
+///         The vault's underlying token (a.k.a. want token) should be the LP
+///         token of the tBTC v2 Saddle pool. This strategy borrows the vault's
+///         underlying token up to the debt limit configured for this strategy
+///         in the vault. In order to make the profit, the strategy deposits
+///         the borrowed tokens into the tBTC v2 Saddle reward pool. Depositing
+///         tokens in the reward pool generates KEEP token rewards.
+///         The financial outcome is settled upon a call of the `harvest` method
+///         (BaseStrategy.sol). Once that call is made, the strategy gets the
+///         KEEP token rewards from Saddle reward pool. These reward tokens are
+///         then used to buy wBTC via a decentralized exchange. At the end, the
+///         strategy takes acquired wBTC and deposits them to the Saddle tBTC v2
+///         pool. This way it obtains new LP tokens the vault is interested in,
+///         and makes the profit in result. At this stage, the strategy may
+///         repay some debt back to the vault, if needed. The entire cycle
+///         repeats for the strategy lifetime, so all gains are constantly
+///         reinvested. Worth to flag that the current implementation uses wBTC
+///         as the intermediary token because of its liquidity and ubiquity in
+///         BTC-based Curve pools.
+/// @dev Implementation is based on:
+///      - General Yearn strategy template
+///        https://github.com/yearn/brownie-strategy-mix
+contract SaddleStrategy is BaseStrategy {
+    using SafeERC20 for IERC20;
+    using Address for address;
+    using SafeMath for uint256;
+
+    // Address of the KEEP token contract.
+    address public constant keep = 0x85Eee30c52B0b379b046Fb0F85F4f3Dc3009aFEC;
```

Comment:
I wonder if we shouldn't make all the token names capital letters across all the contracts. I think this is a standard. cc @lukasz-zimnoch

Reply:
Up to you guys. Both versions work for me.

Reply:
@dimpar:
Should we make all the letters capital, like:
`keep` -> `KEEP`
`weth` - > `WETH` ?

Should the contract addresses stay unchanged:
`tbtcSaddlePoolSwap`
`tbtcSaddleLPRewards`
`uniswap`
?

Reply:
yes, only token names change to capital, like keep->KEEP, wbtc->WBTC etc..

Reply:
I believe that in Solidity code we usually do `keepToken`, `wbtcToken`.

Reply:
ok, I'm fine with the version suggested by Piotr

Reply:
Done in https://github.com/keep-network/tbtc-v2/pull/13/commits/90c065d3d25770743b95530c90d9289f89522f74.